layer/tests/tests_combine_union.cpp

```diff
@@ -52,25 +52,29 @@ class TestsUnion : public VkTestFramework {
         ASSERT_EQ(err, VK_SUCCESS);
     }
 
-    static void TearDownTestSuite(){ 
-        inst_builder.reset();
-    };
+    static void TearDownTestSuite() { inst_builder.reset(); };
 };
 
 TEST_F(TestsUnion, Extension) {
     uint32_t count;
     vkEnumerateDeviceExtensionProperties(gpu_profile, nullptr, &count, nullptr);
     std::vector<VkExtensionProperties> device_extensions(count);
     vkEnumerateDeviceExtensionProperties(gpu_profile, nullptr, &count, device_extensions.data());
-
-    ASSERT_STREQ("VK_AMD_device_coherent_memory", device_extensions[0].extensionName);
-    ASSERT_STREQ("VK_EXT_extended_dynamic_state", device_extensions[1].extensionName);
-    ASSERT_STREQ("VK_EXT_shader_demote_to_helper_invocation", device_extensions[2].extensionName);
-    ASSERT_STREQ("VK_KHR_maintenance1", device_extensions[3].extensionName);
-    ASSERT_STREQ("VK_KHR_maintenance2", device_extensions[4].extensionName);
-    ASSERT_STREQ("VK_KHR_maintenance3", device_extensions[5].extensionName);
-    ASSERT_STREQ("VK_KHR_variable_pointers", device_extensions[6].extensionName);
-    ASSERT_STREQ("VK_KHR_portability_subset", device_extensions[7].extensionName);
+    ASSERT_EQ(device_extensions.size(), count);
+    std::array<const char*, 8> expected_extensions = {
+        "VK_AMD_device_coherent_memory", "VK_EXT_extended_dynamic_state", "VK_EXT_shader_demote_to_helper_invocation",
+        "VK_KHR_maintenance1",           "VK_KHR_maintenance2",           "VK_KHR_maintenance3",
+        "VK_KHR_variable_pointers",      "VK_KHR_portability_subset"};
+    for (const auto& extension : device_extensions) {
```

Comment:
And now the tests are a lot longer? 
Erm...

Reply:
This is because the code now doesn't preserve the order the extensions are enumerated in, due to the use of unordered_map instead of vector to store the extension list. I wasn't 100% about this decision because the underlying issue is the need to quickly check if extensions are available when determining the profile capabilities & whatnot. I could have tried to maintain order but that would of required significantly more code, since there would need to be more work to maintain 'order'. Still, the performance improvement using unordered_map was well work it from my validation.